client/network/src/protocol.rs

```diff
@@ -1147,11 +1147,6 @@ where
 					if self.on_sync_peer_disconnected(peer_id).is_ok() {
 						CustomMessageOutcome::SyncDisconnected(peer_id)
 					} else {
-						log::trace!(
-							target: "sync",
-							"Disconnected peer which had earlier been refused by on_sync_peer_connected {}",
-							peer_id
-						);
```

Comment:
Is there a reason why this trace is emitted only in this specific if branch? May be better keep it?

Reply:
Looking at `on_sync_peer_disconnected`, the error is emitted only if we don't find the peer in the list of connected peers. So IMO it's better to keep this debugging logging.

Reply:
I agree, we should keep it.